### PR TITLE
Justin

### PR DESCRIPTION
**Pull request by KoboUCF**

### CODE REVIEW

File: Scheduler/main.c [hunk 1]
```diff
@@ -420,7 +419,81 @@ void runSJF()
...
+         // Get index at which to start looking for next process to run.
+         i = turn = (idxOfCurrent + 1) % processCount;
 
+         // Starting at 'turn', look through all processes for the next one ready.
+         // If none are ready at or after 'turn', look at indices before it as well.
+         do
+         {
+            if (processes[i].isReady)
+            {
+               printProcessSelected(time, &processes[i]);
```
Comment (robinschiro) [inline Scheduler/main.c]: If there is only one ready process for multiple quantums, then the same selection will be printed multiple times. Selection should only be printed when a new process is selected (look at runSFJ() for an example of this)

File: Scheduler/main.c [hunk 2]
```diff
@@ -420,7 +419,81 @@ void runSJF()
...
+      // Update the wait times of ready processes that were not selected.
+      for (i = 0; i < processCount; i++)
+      {
+         if (processes[i].isReady && (i != idxOfCurrent))
+         {
+            processes[i].wait++;
+         }
+      }
+   }
+
```
Comment (robinschiro) [inline Scheduler/main.c:524]: Be sure to check if the current process is finished after iterating through the time steps (same comment I made for Khoa).

File: Scheduler/main.c [hunk 3]
```diff
@@ -420,7 +419,81 @@ void runSJF()
...
+      if (!quantumRemaining || processFinished)
+      {
+         // Get index at which to start looking for next process to run.
+         i = turn = (idxOfCurrent + 1) % processCount;
 
+         // Starting at 'turn', look through all processes for the next one ready.
+         // If none are ready at or after 'turn', look at indices before it as well.
+         do
+         {
+            if (processes[i].isReady)
```
Comment (robinschiro) [inline Scheduler/main.c]: I see that when you're selecting the next process to process, you are iterating through ready processes in the order that they appear in the 'processes' array (that is, in the order that we parsed the file). According to our notes (Lecture 1.3), they should be selected based on the implementation of a "ready" queue. A process is enqueued on the ready_queue when it arrives and when it is deselected. It is dequeued when it is selected.

**Review by robinschiro — CHANGES_REQUESTED** [review]
I found a serious issue in your implementation (look at comment on line 457). You should be using a "ready" queue to keep track of which process to select next. 

The test cases I made (in the "custom-tests" folder on the master branch) should help verify the correctness of your implementation.

File: Scheduler/main.c (hunk 2 above)
Comment (KoboUCF) [reply Scheduler/main.c:524]: Fixed

**Review by robinschiro — APPROVED** [review]
Looks good!

**Pull request merged**